### PR TITLE
docs: make the docs site AI/LLM-friendly (llms.txt, copy/open-in buttons, markdown negotiation)

### PR DESCRIPTION
**Pull request by mscolnick**
## What

Makes the docs site (`marimohub.docs.marimo.io`) friendly to LLMs and AI tooling, three layers:

### 1. Static generation — `vitepress-plugin-llms`
Every build now emits into `.vitepress/dist`:
- `llms.txt` — sidebar-driven index with absolute URLs ([llmstxt.org](https://llmstxt.org) spec)
- `llms-full.txt` — all docs concatenated
- a raw-markdown twin next to every page (`/storage.md`, `/deploying.md`, …), with `@include` partials and the `deploying/README.md` rewrite resolved
- `sitemap.xml` (new) and hidden per-page LLM hints pointing at the `.md` twins

### 2. Page-actions dropdown (Vue theme component)
`PageActions.vue`, injected via the `doc-before` layout slot (doc pages only):
- **Copy as Markdown** — fetches the page's `.md` twin (DOM-text fallback in dev)
- **Open in Claude** / **Open in ChatGPT** — deep links whose prompt points at the `.md` URL

URL mapping lives in a pure helper (`mdUrl.ts`) with vitest coverage. (Codex has no verified prompt deep-link URL, so ChatGPT is the stand-in; adding entries later is one array item.)

### 3. `Accept: text/markdown` content negotiation
`apps/docs/middleware.ts` (Vercel Edge Middleware, ported from the marimo docs) re-serves the `.md` twin when the client sends `Accept: text/markdown`, with `Vary: Accept` + CORS; `vercel.json` sets markdown content-type headers.

## Bonus fix
The live site 404s on every clean URL today (`/getting-started` → 404, `/getting-started.html` → 200): VitePress has `cleanUrls: true` but the Vercel project didn't. `vercel.json` `cleanUrls: true` fixes it.

## Verification
Verified end-to-end on a preview deployment (via a protection-bypass automation secret):
- `curl -H 'Accept: text/markdown' …/storage` → markdown twin, `text/markdown`, `Vary: Accept`; plain curl → HTML
- `/deploying/` → flattened `deploying.md`; `/` → `index.md`
- `llms.txt`, `llms-full.txt`, `.md` twins all 200 with correct content types
- `/getting-started` → 200 (clean-URL fix)
- `pnpm check`, `pnpm test`, `pnpm build` green

The preview also caught a real edge case: on SSO-protected previews the middleware's twin-fetch got the login interstitial — fixed by forwarding the bypass header and guarding on the fetched content-type.

---
## Summary by cubic
Make the docs site LLM-friendly by emitting markdown twins and llms.txt, adding copy/open-in actions, and serving markdown via content negotiation. Also fixes 404s on clean URLs.

- **New Features**
  - Generate `llms.txt`, `llms-full.txt`, per-page `.md` twins, and `sitemap.xml` via `vitepress-plugin-llms`.
  - Add page actions: Copy as Markdown, Open in Claude, Open in ChatGPT (deep links target the `.md` twin). Menu is keyboard-accessible (Arrow/Home/End, Escape/Tab), and copy falls back when `clipboard.writeText` fails; warns and falls back to rendered text if the twin isn’t available.
  - Edge middleware serves the `.md` twin when `Accept: text/markdown` is sent, with `Vary: Accept` and CORS.
  - `mdUrlForPath` helper maps routes to twins, with tests.

- **Bug Fixes**
  - Enable `cleanUrls` in Vercel to stop 404s on extensionless paths.

<sup>Written for commit e0cc3b5bc3b7def89bb65eddb65cb8b83c9b7143. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/marimo-team/marimohub/pull/14?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: apps/docs/.vitepress/theme/components/PageActions.vue [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+	}
+	return null;
+}
+
+async function copyText(text: string): Promise<void> {
+	if (navigator.clipboard && window.isSecureContext) {
+		await navigator.clipboard.writeText(text);
+		return;
+	}
+	const textarea = document.createElement('textarea');
```
[the remark this answers is not in the document]
Comment (mscolnick) [reply apps/docs/.vitepress/theme/components/PageActions.vue:42]: Fixed in e0cc3b5b — `copyText` now try/catches the Clipboard API write and falls back to the textarea path when it rejects (e.g. gesture expired during the twin fetch), so the copy still lands and the "Copied!" feedback fires.

File: apps/docs/.vitepress/theme/components/PageActions.vue [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+	textarea.style.position = 'fixed';
+	textarea.style.opacity = '0';
+	document.body.appendChild(textarea);
+	textarea.select();
+	document.execCommand('copy');
+	textarea.remove();
+}
+
+async function copyMarkdown(): Promise<void> {
+	const markdown = (await fetchMarkdown()) ?? document.querySelector('.vp-doc')?.textContent ?? '';
```
[the remark this answers is not in the document]
Comment (mscolnick) [reply apps/docs/.vitepress/theme/components/PageActions.vue]: Fixed in e0cc3b5b — the fallback now logs a `console.warn` when the markdown twin is unavailable and rendered text is copied instead, so the behavior is observable in dev.

File: apps/docs/.vitepress/theme/components/PageActions.vue [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+				fill="none"
+				stroke="currentColor"
+				stroke-width="2"
+				aria-hidden="true"
+			>
+				<path d="m6 9 6 6 6-6" />
+			</svg>
+		</button>
+
+		<div v-if="open" class="page-actions-menu" role="menu">
```
[the remark this answers is not in the document]
Comment (mscolnick) [reply apps/docs/.vitepress/theme/components/PageActions.vue]: Fixed in e0cc3b5b — implemented the menu-button keyboard model: opening focuses the first item, ArrowUp/ArrowDown wrap through items, Home/End jump, Escape closes and returns focus to the toggle, Tab closes; menu items have a visible focus style.
